File: android/src/main/java/org/axen/flutter/glide/provider/BitmapImageProvider.java
package org.axen.flutter.glide.provider;

import android.content.Context;
import android.graphics.Bitmap;

import androidx.annotation.NonNull;

import com.bumptech.glide.Glide;
import com.bumptech.glide.RequestBuilder;

public final class BitmapImageProvider extends AbstractProvider<Bitmap> {

    public BitmapImageProvider(@NonNull Context context) {
        super(context);
    }

    @NonNull
    protected RequestBuilder<Bitmap> build() {
        return Glide.with(this.context).asBitmap();
    }
}

File: android/src/main/kotlin/org/axen/flutter/glide/provider/BitmapImageProvider.kt
package org.axen.flutter.glide.provider

import android.content.Context
import android.graphics.Bitmap
import com.bumptech.glide.Glide
import com.bumptech.glide.RequestBuilder
import org.axen.flutter.glide.common.AbstractProvider

class BitmapImageProvider(context: Context): AbstractProvider<Bitmap>(context) {

    override fun build(): RequestBuilder<Bitmap> {
        return Glide.with(context).asBitmap()
    }
}